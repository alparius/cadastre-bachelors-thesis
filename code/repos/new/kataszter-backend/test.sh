#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
