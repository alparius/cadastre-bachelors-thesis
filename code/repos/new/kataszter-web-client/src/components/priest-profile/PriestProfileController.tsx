import React from "react";
import { FormattedMessage as FM } from "react-intl";
import { Divider, Form, Header, Menu, Responsive } from "semantic-ui-react";

import PriestProfileTab from "../../data/enums/PriestProfileTab";

interface IPriestProfileControllerProps {
    currentTab: string;
    editMode: boolean;
    setCurrentTab(newCurrentTab: string): void;
}

const PriestProfileController: React.FC<IPriestProfileControllerProps> = (props: IPriestProfileControllerProps) => {
    const { currentTab, setCurrentTab, editMode } = props;

    const handleCurrentTabChange = (e: any, { value }: any) => setCurrentTab(value);

    return (
        <>
            {editMode && <Responsive as={Divider} maxWidth={768} style={{ marginTop: "6em" }} />}
            <Form>
                <Header as="h1" className="controllerHeader">
                    <FM id="profile.profilePriest" />
                </Header>

                <Menu fluid vertical secondary color="violet" size="huge">
                    <Menu.Item
                        id="general"
                        value={PriestProfileTab.general}
                        active={currentTab === PriestProfileTab.general}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="profile.generalData" />
                    </Menu.Item>
                    <Menu.Item
                        id="family"
                        value={PriestProfileTab.family}
                        active={currentTab === PriestProfileTab.family}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.family" />
                    </Menu.Item>
                    <Menu.Item
                        id="qualification"
                        value={PriestProfileTab.qualification}
                        active={currentTab === PriestProfileTab.qualification}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.qualification" />
                    </Menu.Item>
                    <Menu.Item
                        id="placements"
                        value={PriestProfileTab.placements}
                        active={currentTab === PriestProfileTab.placements}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.placements" />
                    </Menu.Item>
                    <Menu.Item
                        id="occupations"
                        value={PriestProfileTab.occupations}
                        active={currentTab === PriestProfileTab.occupations}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.occupations" />
                    </Menu.Item>
                    <Menu.Item
                        id="disciplinaryMatters"
                        value={PriestProfileTab.disciplinaryMatters}
                        active={currentTab === PriestProfileTab.disciplinaryMatters}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.disciplinaryMatters" />
                    </Menu.Item>
                    <Menu.Item
                        id="pathOfLife"
                        value={PriestProfileTab.pathOfLife}
                        active={currentTab === PriestProfileTab.pathOfLife}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.pathOfLife" />
                    </Menu.Item>
                    <Menu.Item
                        id="references"
                        value={PriestProfileTab.references}
                        active={currentTab === PriestProfileTab.references}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.references" />
                    </Menu.Item>
                    <Menu.Item
                        id="pictures"
                        value={PriestProfileTab.pictures}
                        active={currentTab === PriestProfileTab.pictures}
                        disabled
                        // onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.pictures" />
                    </Menu.Item>
                    <Menu.Item
                        id="files"
                        value={PriestProfileTab.files}
                        active={currentTab === PriestProfileTab.files}
                        disabled
                        // onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.files" />
                    </Menu.Item>
                    <Menu.Item
                        id="notes"
                        value={PriestProfileTab.notes}
                        active={currentTab === PriestProfileTab.notes}
                        onClick={handleCurrentTabChange}
                    >
                        <FM id="menu.notes" />
                    </Menu.Item>
                </Menu>
            </Form>
        </>
    );
};

export default PriestProfileController;
