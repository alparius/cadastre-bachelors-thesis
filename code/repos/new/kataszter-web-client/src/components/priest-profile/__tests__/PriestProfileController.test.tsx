/*===================
  PriestProfileController.unit.test.tsx
 ===================*/
import React from "react";

import { createComponentWithIntl, mountWithIntl } from "../../../util/test/test-wrapper";
import PriestProfileController from "../PriestProfileController";

const { ...handlers } = {
    setCurrentTab: jest.fn()
};

describe("PriestProfileController element", () => {
    afterEach(() => {
        handlers.setCurrentTab.mockClear();
    });

    it("should render the form with menu without upper shifting when editMode=false", () => {
        const container = createComponentWithIntl(<PriestProfileController editMode={false} currentTab="general" {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should render the form with menu with upper shifting when editMode=ture", () => {
        const container = createComponentWithIntl(<PriestProfileController editMode currentTab="general" {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call the handleCurrentTabChange function when clicked on menu tabs", () => {
        const container = mountWithIntl(<PriestProfileController editMode={false} currentTab="general" {...handlers} />);
        container
            .find("#family")
            .at(0)
            .simulate("click");
        expect(handlers.setCurrentTab).toHaveBeenCalledTimes(1);
        expect(handlers.setCurrentTab).toHaveBeenCalledWith("family");
    });
});
