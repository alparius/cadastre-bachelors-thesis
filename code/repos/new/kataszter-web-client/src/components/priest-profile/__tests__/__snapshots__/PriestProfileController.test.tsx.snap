// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriestProfileController element should render the form with menu with upper shifting when editMode=ture 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <h1
    className="ui header controllerHeader"
  >
    Priest profile
  </h1>
  <div
    className="ui violet huge fluid secondary vertical menu"
  >
    <a
      className="active item"
      id="general"
      onClick={[Function]}
      value="general"
    >
      General data
    </a>
    <a
      className="item"
      id="family"
      onClick={[Function]}
      value="family"
    >
      Family
    </a>
    <a
      className="item"
      id="qualification"
      onClick={[Function]}
      value="qualification"
    >
      Qualifications
    </a>
    <a
      className="item"
      id="placements"
      onClick={[Function]}
      value="placements"
    >
      Placements
    </a>
    <a
      className="item"
      id="occupations"
      onClick={[Function]}
      value="occupations"
    >
      Occupations
    </a>
    <a
      className="item"
      id="disciplinaryMatters"
      onClick={[Function]}
      value="disciplinaryMatters"
    >
      Disciplinary matters
    </a>
    <a
      className="item"
      id="pathOfLife"
      onClick={[Function]}
      value="pathOfLife"
    >
      Path of life
    </a>
    <a
      className="item"
      id="references"
      onClick={[Function]}
      value="references"
    >
      References
    </a>
    <div
      className="disabled item"
      id="pictures"
      onClick={[Function]}
      value="pictures"
    >
      Pictures
    </div>
    <div
      className="disabled item"
      id="files"
      onClick={[Function]}
      value="files"
    >
      Files
    </div>
    <a
      className="item"
      id="notes"
      onClick={[Function]}
      value="notes"
    >
      Notes
    </a>
  </div>
</form>
`;

exports[`PriestProfileController element should render the form with menu without upper shifting when editMode=false 1`] = `
<form
  className="ui form"
  onSubmit={[Function]}
>
  <h1
    className="ui header controllerHeader"
  >
    Priest profile
  </h1>
  <div
    className="ui violet huge fluid secondary vertical menu"
  >
    <a
      className="active item"
      id="general"
      onClick={[Function]}
      value="general"
    >
      General data
    </a>
    <a
      className="item"
      id="family"
      onClick={[Function]}
      value="family"
    >
      Family
    </a>
    <a
      className="item"
      id="qualification"
      onClick={[Function]}
      value="qualification"
    >
      Qualifications
    </a>
    <a
      className="item"
      id="placements"
      onClick={[Function]}
      value="placements"
    >
      Placements
    </a>
    <a
      className="item"
      id="occupations"
      onClick={[Function]}
      value="occupations"
    >
      Occupations
    </a>
    <a
      className="item"
      id="disciplinaryMatters"
      onClick={[Function]}
      value="disciplinaryMatters"
    >
      Disciplinary matters
    </a>
    <a
      className="item"
      id="pathOfLife"
      onClick={[Function]}
      value="pathOfLife"
    >
      Path of life
    </a>
    <a
      className="item"
      id="references"
      onClick={[Function]}
      value="references"
    >
      References
    </a>
    <div
      className="disabled item"
      id="pictures"
      onClick={[Function]}
      value="pictures"
    >
      Pictures
    </div>
    <div
      className="disabled item"
      id="files"
      onClick={[Function]}
      value="files"
    >
      Files
    </div>
    <a
      className="item"
      id="notes"
      onClick={[Function]}
      value="notes"
    >
      Notes
    </a>
  </div>
</form>
`;
